File: solutions/python/arrays-hashing/two-sum.py
# LeetCode 1 — Two Sum (Easy)
# Category: Arrays & Hashing · Approach: Hash Map
# Time: O(n) | Space: O(n)
# Source: https://leetcode.com/problems/two-sum/

def two_sum(nums: list[int], target: int) -> list[int]:
    seen: dict[int, int] = {}
    for i, x in enumerate(nums):
        need = target - x
        if need in seen:
            return [seen[need], i]
        seen[x] = i
    return []

File: solutions/typescript/arrays-hashing/two-sum.ts
// LeetCode 1 — Two Sum (Easy)
// Category: Arrays & Hashing · Approach: Hash Map
// Time: O(n) | Space: O(n)
// Source: https://leetcode.com/problems/two-sum/

function twoSum(nums: number[], target: number): number[] {
  const seen = new Map<number, number>();
  for (let i = 0; i < nums.length; i++) {
    const need = target - nums[i];
    if (seen.has(need)) return [seen.get(need)!, i];
    seen.set(nums[i], i);
  }
  return [];
}
